src/common/meta/src/key/datanode_table.rs

```diff
@@ -89,6 +96,8 @@ pub struct DatanodeTableValue {
     pub engine: String,
     #[serde(default)]
     pub region_storage_path: String,
+    #[serde(default)]
+    pub region_options: HashMap<String, String>,
```

Comment:
I don't think `region_options`(as well as `engine` and `region_storage_path`) is best placed in `DatanodeTableValue`. These region attributes has little to do with the placement of regions among Datanode, which is the intention of `DatanodeTableValue` in the first place. 

It looks to me that doing so is trying to give us the direct convenience when figuring out the regions to open in Datanode. That's fine. But I think a more proper way to do it is "indirect": move these region attributes into a region id indexed `RegionInfoValue`. That might introduce extra efforts when opening the regions, but gain more flexibility, and less counterintuitive.

Reply:
The key problem is to reduce requests sent to etcd.

Reply:
@WenyXu @MichaelScofield Have we reached an agreement here?